### DISCUSSION TITLE
Better UX by implementing iOS design guidelines

### DISCUSSION THREAD

**Discussion by sanketpath** [wallabag/ios-app#388] [Ideas]
Ola 👋 

I've been using Wallabag since past month or two and I certainly see a great potential in this app. I'm a designer by profession and would _love_ to contribute in improving the user experience of this app. However, I'm not a developer and wondering if anyone else is willing to implement the design changes.

I don't mean to take the entire app through redesign. I don't have enough context into how app is being used currently. However, I do see some clear improvements if we can correct a few Swift UI implementation as per the [iOS design guidelines](https://developer.apple.com/design/human-interface-guidelines/guidelines/overview/).

Let me know if anyone is interested. Thanks!

_P.s. I hope 'Issues' is the right place to discuss this. If not, please direct me to the right channel or page._

**Comment by bourvill** [comment]
Hello, I migrated the issue into discussion.

It’s a good idea, let’s discuss about your idea here, and we will create related issue for each idea or change request.

Ps: at this time, I’m only the developer on this repository, and no more spare time to code actively

**Reply by sanketpath** [reply]
Thanks @bourvill for your response and moving this into right place.

I truly appreciate the efforts you've put into this 🙌 
> at this time, I’m only the developer on this repository

I'll document a couple of ideas I've over this week or weekend and post it here. Stay tuned 😉

**Comment by sanketpath** [comment]
@bourvill Here's a first set of ideas. Also attaching rough designs below.

**Section: Nav bar**

1. **Changing the menu-icon** 
The icon that we currently use is a representation of a text formatting option — order list — something that we commonly see in Apple Notes, Google Docs, Microsoft Word and so on…  Here’s the [standard menu icon](https://feathericons.com/?query=menu) we can use to represent the menu aka Hamburger Menu. 

2. **Changing the title of home page** 
The title is currently ‘Articles’ which is slightly misleading. In Wallabg, my hypothesis is, people collect more than articles. Example, videos to watch later, podcasts to save for later, song playlist yo check out for later etc. Additionally, the first page is the page everyone lands on. Hence, calling it ‘Home’ makes more sense. Alternatively, we can call it ’Saved items’ as well.

3. **Actions on the right side**
- Removing `Refresh` button and add swipe-down-to-refresh interaction – something that's a standard design pattern used across products.
- Moving `search` in the nav bar to as it's one of the primary actions. This also allows us to create better spacing for the segmented control.
- Adding `More` menu (with 3 dots) to include additional actions. Instead of keeping `Add entry` in the left-menu-bar, it's best to keep it in the context menu (in the design) that follows Apple Guidelines.
- Move `Show Playlist` option under `More` menu. My hypothesis is, `Playlist` isn't the frequently used option.
- In future, we can think if `Bulk Edit` feature as well. For now, please ignore it.

Let me know what you think. I'm happy to work on the final designs and share it here so you can implement them :)

<img width="1624" alt="Screenshot 2022-10-18 at 20 16 42" src="https://user-images.githubusercontent.com/3312854/196469431-0b45333c-7497-404e-b4e6-e9b9e08ad4c2.png">

**Comment by bourvill** [comment]
Thank you very much for your very constructive and very detailed feedback.

I will try to add the issues following your feedback to add them to the application.

**Reply by sanketpath** [reply]
Awesome! More than happy to work closely with you on this one. I can help more with quick reviews before final deployment.
